Assets/Scripts/Tank/TankService.cs

```diff
@@ -1,37 +1,97 @@
-﻿using UnityEngine;
+﻿using System.Collections.Generic;
+using UnityEngine;
 public class TankService : MonoSingleton<TankService>
 {
-    public TankView tankView;
     private Camera mainCamera;
-    private TankModel tankModel;
-    private TankController tankController;
-    protected Joystick stick;
+
+    [SerializeField]
+    private List<EnemyController> enemyList;
+    [SerializeField]
+    private List<Transform> spawnPositions;
+    [SerializeField]
+    private SpawnerObject spawnerObject;
+    public List<TankScriptableObject> enemiesToGenerate;
+    public Transform enemyParent;
+    public EnemyController redTank;
```

Comment:
What if there are 1000 tanks? Are you going to have that many variables? A better way is to make each tank scriptable object contain its own prefab that needs to be instantiated

Reply:
done